src/views/Assignment/AssignmentDetailView.vue

```diff
@@ -148,6 +156,63 @@ export default class AssignmentView extends Vue {
       });
   }
 
+  async deleteAssignment(): Promise<void> {
+    this.garnBarnAPICaller?.v1.assignments
+      .delete(this.assignmentId)
+      .then((apiResponse) => {
+        this.informDialogBox.show({
+          dialogBoxContent: {
+            title: "Assignment deleted",
+            content: `Your assignment has been deleted.`,
+          },
+          dialogBoxActions: [
+            {
+              buttonContent: "Ok",
+              buttonClass: "md-secondary",
+              onClick: async () => {
+                this.informDialogBox.dismiss();
+              },
+            },
+          ],
+        });
+      })
+      .catch((e) => {
+        this.informDialogBox.show({
+          dialogBoxContent: {
+            title: "Error",
+            content: e.message,
+          },
+        });
+      });
+  }
+
+  confirmDelete(): void {
+    this.informDialogBox.show({
+      dialogBoxContent: {
+        title: "Confirm Delete?",
+        content: `This action can't be undone, Are you sure?`,
+      },
+      dialogBoxActions: [
+        {
+          buttonContent: "Confirm",
+          buttonClass: "md-primary",
+          onClick: (): void => {
+            this.deleteAssignment();
+            this.informDialogBox.dismiss();
+            this.$router.back();
+          },
```

Comment:
If the `this.deleteAssignment()` throw an error, I think it would be better if we wait to see the result before popping the page back